relayer/src/chain/cosmos.rs

```diff
@@ -779,11 +788,19 @@ impl ChainEndpoint for CosmosSdkChain {
         crate::telemetry!(query, self.id(), "query_upgraded_client_state");
 
         // Query for the value and the proof.
-        let tm_height =
+        let upgrade_height =
             Height::try_from(request.height.revision_height).map_err(Error::invalid_height)?;
+        let query_height = Height::try_from(
```

Comment:
Wouldn't it work if we do `Height::decrement` ?

https://github.com/informalsystems/ibc-rs/blob/ad2d8c59083c4670c4ed0d8b6d446040fa7044f1/modules/src/core/ics02_client/height.rs#L65-L67

Reply:
`Height` in this file is actually `tendermint::block::height::Height`, and it only has an `increment()`. I thought about renaming it to `TMHeight`, but decided against to reduce the number of merge conflicts with my other Height zero PR. I can address it there though.

The alternative way to write it is:
```rust
let query_height = Height::try_from(
            request
                .height
                .decrement()
                .expect("height overflow")
                .revision_height,
        )
        .map_err(Error::invalid_height)?;
```
I personally prefer the current one, since although not ideal, at least it's clearer that we're doing `upgrade_height - 1`.

Reply:
You're converting from `ibc::Height` to `tendermint::block::Height` in the line above, no?

```rust
Height::try_from(request.height.revision_height)
```

but this is a minor concern. It would improve readability if we could simplify the snipper, but feel free to ignore if the solution is not obvious.